### ISSUE TITLE
How Can I get more fps with darknet?

### ISSUE THREAD

**Issue by passion3394** [AlexeyAB/darknet#713]
How Can I get more fps with darknet? I use darknet on Jetson TX2, and got only 3.3 fps when run yolov3.

**Comment by AlexeyAB** [comment]
Or try to use `yolov2-tiny.cfg` and `yolov2-tiny.weights`: https://github.com/AlexeyAB/darknet#pre-trained-models-for-different-cfg-files-can-be-downloaded-from-smaller---faster--lower-quality

Or buy GPU nVidia Titan V and compile with `CUDNN_HALF=1`: https://github.com/AlexeyAB/darknet/issues/407#issue-300029052

----

If you are using **Jetson TX2** then do this:

1. Run `jetson-clocks.sh` and `sudo nvpmodel -m 0` https://github.com/AlexeyAB/darknet/issues/436#issuecomment-371786781
More info: http://www.jetsonhacks.com/2017/03/25/nvpmodel-nvidia-jetson-tx2-development-kit/

2. Change these lines to `width=288 height=288`: https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/cfg/yolov3.cfg#L8-L9

3. What param do you use here? https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/Makefile#L1-L2

4. Un-comment this line and re-compile: https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/Makefile#L36

And comment these lines: https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/Makefile#L15-L19

**Comment by passion3394** [comment]
Thanks very much. I use GPU = 1 and CUDNN = 1.
I will try to modify the above configs which you have mentioned, then I will post the new fps here.

**Comment by AshleyRoth** [comment]
@passion3394, hi there! Do you try with a new params? Wanna know information about fps, it's success for you? thanks

**Comment by passion3394** [comment]
@AshleyRoth Not yet. I will do it later.

**Comment by passion3394** [comment]
@AlexeyAB @AshleyRoth sorry for long time no reply.
First, I use the proposes you gave, the fps get 5.1 fps from the original 3.2 fps 
Then I modify the following TX2 command:
`./darknet detector demo cfg/coco.data cfg/yolov3.cfg yolov3.weights "nvcamerasrc ! video/x-raw(memory:NVMM), width=(int)1280, height=(int)720, format=(string)I420, framerate=(fraction)30/1 ! nvvidconv flip-method=0 ! video/x-raw, format=(string)BGRx ! videoconvert ! video/x-raw, format=(string)BGR ! appsink"`

TO
`./darknet detector demo cfg/coco.data cfg/yolov3.cfg yolov3.weights "nvcamerasrc ! video/x-raw(memory:NVMM), width=(int)640, height=(int)360, format=(string)I420, framerate=(fraction)30/1 ! nvvidconv flip-method=0 ! video/x-raw, format=(string)BGRx ! videoconvert ! video/x-raw, format=(string)BGR ! appsink"`

The difference of the two code segment is that width modifed from 1280 to 640 and height modified from 720 to 360. In my sense, I think the fps should be about 10,**but it is still 5.x** Why?

**Comment by AlexeyAB** [comment]
@passion3394 

> ./darknet detector demo cfg/coco.data cfg/yolov3.cfg yolov3.weights "nvcamerasrc ! video/x-raw(memory:NVMM), width=(int)1280, height=(int)720
* `width=(int)1280, height=(int)720` - this is the size of image (video frame), and it has a minimal effect on performance, because it is already optimized in this fork. Any size image will be resized to the network size (width and height in the cfg-file), so it doesn't matter what size of image.

* the Neural Network has the most computation cost. Speed depends on its width, height: https://github.com/AlexeyAB/darknet/blob/0948df52b850b908e7a74cb589d19fa29eb30368/cfg/yolov3.cfg#L8-L9
    and depends on the architecture of network `yolov3.cfg`, `yolov2.cfg`, `yolov2-tiny.cfg`, ...

**Comment by AlexeyAB** [comment]
@passion3394 Also do you get any advantage by runing this commands on TX2?
`jetson-clocks.sh`
`sudo nvpmodel -m 0`

**Comment by passion3394** [comment]
@AlexeyAB 
After
Run jetson-clocks.sh and sudo nvpmodel -m 0
and
Change width=416 height=416 to width=288 height=288
I get 5.1 fps compared to 3.2fps
